src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -1047,12 +1048,20 @@ inlineToHtml opts inline = do
                                               strToHtml "’")
                               DoubleQuote -> (strToHtml "“",
                                               strToHtml "”")
-                        in  if writerHtmlQTags opts
+
+                        in if writerHtmlQTags opts
                                then do
                                  modify $ \st -> st{ stQuotes = True }
-                                 H.q `fmap` inlineListToHtml opts lst
+                                 H.q `fmap` inlineListToHtml opts lst' >>= withAttr
                                else (\x -> leftQuote >> x >> rightQuote)
-                                    `fmap` inlineListToHtml opts lst
+                                    `fmap` inlineListToHtml opts lst'
+                          where
+                            lst' = maybe lst (\cs -> cs ++ tail lst) (snd <$> citeSpan)
```

Comment:
We should generally avoid using partial functions like `tail` -- consider using an alternative, perhaps one of those defined in Safe.

Reply:
`tail` is safe to use here because the code will only run if `listToMaybe` returned `Just` which means there are at least one element in `lst`. Nonetheless I see your point. Would you like me to change it still?

Reply:
> yeah, we've had cases when a later refactoring change the thing around the partial function...

Yeah that is a good point. This will be changed when I rather pattern match directly (as mentioned below).

Reply:
Yes, it's good practice just to avoid using these partial functions; then you know you won't get in trouble.